state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -120,6 +123,17 @@ pub enum DepositIgnoredReason {
 	/// The deposit was ignored because the amount provided was not high enough to pay for the fees
 	/// required to process the requisite transactions.
 	NotEnoughToPayFees,
+	TransactionTainted,
+}
+
+/// Holds information about a tainted transaction.
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo, CloneNoBound)]
+#[scale_info(skip_type_params(T, I))]
+pub struct TaintedTransactionDetails<T: Config<I>, I: 'static> {
+	pub refund_address: Option<ForeignChainAddress>,
```

Comment:
Until refund parameters are optional, this has to stay optional. We can change it after this is done.

Reply:
But we can't refund anything if there is no refund address? 

Doesn't it make more sense to make it required here, and handle the case where we don't have it before storing this value?

Reply:
I don't believe it doesn't make any difference other than having to deal with less complex code if we go with the option variant. In the point of the swap flow where we have (or have not) the refund address, it's already too late to do anything else than saving what we have and exit the swap logic. What do you want to save if, instead, if there is just no address? The solution is to make the refund parameters a requirement, so this can not happen, but since this is not done, yet we go with an option here and replace it after @msgmaxim has solved this in his PR.

Reply:
Yeah, my preference is that we keep the option in this PR. Then once I change the interface to always require refund address, it can update this code too (and the code won't get compiled if I forget to do this).